test/node/bigint.test.ts

```diff
@@ -367,7 +364,7 @@ describe('BSON BigInt support', function () {
       }
     });
 
-    describe('relaxed double input where double is outside of int32 range and useBigInt64 is true', function () {
+    describe('relaxed double input where double is outside of int32 range and useBigInt64 is true', function() {
```

Comment:
We can add a test that checks BigInt usage is all behind gated logic. Take a look at the example where we test the library with and without a crypto global here: https://github.com/mongodb/js-bson/blob/main/test/node/byte_utils.test.ts#L603

We can load the BSON library with BigInt set to undefined, that doesn't cover using trailing n syntax but it's something

Reply:
Would it be appropriate to do this in another PR as just removing the `BigInt` field from the `globalThis` object isn't enough to be sure that our BigInt work doesn't break anything in environments that don't support BigInts. We'd have to go through the ES BigInt spec and make sure that we're not using any of the associated functionality and that seems outside the scope of this PR.

Reply:
sure, let's file a follow up